crates/nu-command/src/commands/str_/to_datetime.rs

```diff
@@ -8,14 +8,48 @@ use nu_protocol::{
 use nu_source::{Tag, Tagged};
 use nu_value_ext::ValueExt;
 
-use chrono::{DateTime, FixedOffset, LocalResult, Offset, TimeZone};
+use chrono::{DateTime, FixedOffset, Local, LocalResult, Offset, TimeZone, Utc};
 
 #[derive(Deserialize)]
 struct Arguments {
+    timezone: Option<Tagged<String>>,
+    offset: Option<Tagged<i16>>,
     format: Option<Tagged<String>>,
     rest: Vec<ColumnPath>,
 }
 
+// In case it may be confused with chrono::TimeZone
+#[derive(Clone)]
+enum Zone {
+    Utc,
+    Local,
+    East(u8),
+    West(u8),
+    Error, // we want the nullshell to cast it instead of rust
+}
+
+impl Zone {
+    fn new(i: i16) -> Self {
+        if i.abs() <= 12 {
+            // guanranteed here
+            if i >= 0 {
+                Self::East(i as u8) // won't go out of range
+            } else {
+                Self::West(-i as u8) // same here
+            }
+        } else {
+            Self::Error // Out of range
+        }
+    }
+    fn from_string(s: String) -> Self {
```

Comment:
you may be able to simplify this by making your incoming string lowercase and then just using `utc | u` and `local | l` in the arms.

Reply:
Thanks, I will fix it.